fili-core/src/main/java/com/yahoo/bard/webservice/web/DataApiRequest.java

```diff
@@ -296,6 +309,87 @@ public DataApiRequest(
     }
 
     /**
+     * To check whether dateTime column request is first one in the sort list or not.
+     *
+     * @param sortColumns  map of columns and its direction
+     *
+     * @return True if dateTime column is first one in the sort list. False otherwise
+     */
+    protected Boolean isDateTimeFirstSortField(Map<String, SortDirection> sortColumns) {
+        if (sortColumns != null) {
+            List<String> columns = new ArrayList(sortColumns.keySet());
+            return columns.get(0).equals(DATE_TIME_STRING);
+        }
+        else {
+            return false;
+        }
+    }
+
+    /**
+     * Method to remove the dateTime column from map of columns and its direction.
+     *
+     * @param sortColumns  map of columns and its direction
+     *
+     * @return  Map of columns and its direction without dateTime sort column
+     */
+    protected Map<String, SortDirection> removeDateTimeSortColumn(Map<String, SortDirection> sortColumns) {
+        if (sortColumns != null && sortColumns.containsKey(DATE_TIME_STRING)) {
+            sortColumns.remove(DATE_TIME_STRING);
+            return sortColumns;
+        }
+        else {
+            return sortColumns;
+        }
+    }
+
+    /**
+     * Method to generate DateTime sort column from the map of columns and its direction.
+     *
+     * @param sortColumns  map of columns and its direction
+     *
+     * @return Instance of OrderByColumn for dateTime
+     */
+    protected Optional<OrderByColumn> generateDateTimeSortColumn(Map<String, SortDirection> sortColumns) {
+
+        if (sortColumns != null && sortColumns.containsKey(DATE_TIME_STRING)) {
+            if (!isDateTimeFirstSortField(sortColumns)) {
+                LOG.debug(DATE_TIME_SORT_VALUE_INVALID.logFormat());
+                throw new BadApiRequestException(DATE_TIME_SORT_VALUE_INVALID.format());
+            } else {
+                return Optional.of(new OrderByColumn(DATE_TIME_STRING, sortColumns.get(DATE_TIME_STRING)));
+            }
+
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    /**
+     * Method to convert sort list to column and direction map.
+     *
+     * @param sorts  String of sort columns
+     *
+     * @return Map of columns and its direction
+     */
+    protected Map<String, SortDirection> generateSortColumns(String sorts) {
+        if (sorts != null && !sorts.isEmpty()) {
+            return Arrays.asList(sorts.split(",")).stream()
+                    .map(e -> Arrays.asList(e.split("\\|")))
+                    .collect(Collectors.toMap(
+                            e -> e.get(0), e -> getSortDirection(e),
+                                    (u, v) -> {
+                                        LOG.debug(SORT_REQUESTED_ON_DUPLICATE_COLUMN.logFormat(u));
```

Comment:
I don't think we can actually make this error check without much more indication in the change log, nor am I sure we want to add it... Previously, we were happily allowing the same sort to show up more than once (I think), and were probably doing whatever Druid did for sorting (so probably either first or last wins). Since this would technically be a breaking API change, we can't just add it in here, in case someone is relying on how this works today.